gateway/src/node/mod.rs

```diff
@@ -565,21 +571,28 @@ impl<St> Gateway<St> {
         // Once this is a bit more mature, make this a commandline flag instead of a compile time
         // flag
         #[cfg(all(feature = "wireguard", target_os = "linux"))]
-        let wg_api = self.start_wireguard(shutdown.fork("wireguard")).await.ok();
+        let wg_api = self
+            .start_wireguard(shutdown.fork("wireguard"))
+            .await
+            .map_err(|source| GatewayError::StdError { source })?;
 
         #[cfg(all(feature = "wireguard", not(target_os = "linux")))]
         self.start_wireguard(shutdown.fork("wireguard")).await;
 
         info!("Finished nym gateway startup procedure - it should now be able to receive mix and client traffic!");
 
+        info!(
+            "Public key: {:?}",
+            self.identity_keypair.public_key().to_string()
+        );
+
         if let Err(source) = shutdown.wait_for_shutdown().await {
             // that's a nasty workaround, but anyhow errors are generally nicer, especially on exit
             return Err(GatewayError::ShutdownFailure { source });
         }
         #[cfg(all(feature = "wireguard", target_os = "linux"))]
-        if let Some(wg_api) = wg_api {
-            defguard_wireguard_rs::WireguardInterfaceApi::remove_interface(&wg_api)?;
-        }
+        defguard_wireguard_rs::WireguardInterfaceApi::remove_interface(&wg_api)?;
```

Comment:
Yeah, I was looking now at this code and was thinking of something similar for the next PR. But since you've mentioned it, no reason to not do it here. I'll push a commit for it